swift-sdk/Internal/InternalIterableAPI.swift

```diff
@@ -475,14 +486,23 @@ final class InternalIterableAPI: NSObject, PushTrackerProtocol, AuthProvider {
         localStorage.userId = _userId
     }
     
-    private func onLogin() {
+    private func onLogin(_ authToken: String? = nil) {
         ITBInfo()
         
-        if isEitherUserIdOrEmailSet() && config.authDelegate != nil {
+        if let authToken = authToken {
+            self.authManager.setNewTokenAfterUpdateEmail(authToken)
+        } else if isEitherUserIdOrEmailSet() && config.authDelegate != nil {
             requestNewAuthToken()
-        } else {
-            completeUserLogin()
+            return
```

Comment:
We should try to exit only in the beginning of method. Exiting in the middle is a code smell.

Reply:
You can move `completeUserLogin` after line 493 and then add an else clause with `completeUserLogin` added there.